addons/mail/models/mail_thread.py

```diff
@@ -1527,8 +1527,10 @@ def _message_add_suggested_recipient(self, result, partner=None, email=None, rea
             return result
         if partner and partner.email:  # complete profile: id, name <email>
             result[self.ids[0]].append((partner.id, partner.email_formatted, reason))
-        elif partner:  # incomplete profile: id, name
-            result[self.ids[0]].append((partner.id, partner.name or '', reason))
+        elif partner and partner.name: # incomplete profile (missing email): id, name
+            result[self.ids[0]].append((partner.id, partner.name, reason))
+        elif partner:  # incomplete profile (missing email and name) -> skip
+            return result
         else:  # unknown partner, we are probably managing an email address
             result[self.ids[0]].append((False, partner_info.get('full_name') or email, reason))
```

Comment:
Maybe we could simplify
```suggestion
        elif not partner:  # unknown partner, we are probably managing an email address
            result[self.ids[0]].append((False, partner_info.get('full_name') or email, reason))
```
and maybe add a comment explaining we skip empty partners as it has no value for chatter ?